waspc/data/Generator/templates/react-app/src/socket.ts

```diff
@@ -0,0 +1,4 @@
+import { io } from 'socket.io-client'
+import config from './config'
+
+export const socket = io(config.apiUrl, { autoConnect: false })
```

Comment:
I think it is better to only connect when `useSocket` is used (which takes care of connecting by default). Thoughts?

Reply:
Actually, reversed that since handling connection logic across multiple components is not worth the extra complexity.